Makefile

```diff
@@ -2,14 +2,43 @@ BIN_DIR = bin
 export GOPATH ?= $(shell go env GOPATH)
 export GO111MODULE ?= on
 
+LINUX=LINUX
+OSX=OSX
+WINDOWS=WIN32
+OSFLAG :=
+ifeq ($(OS),Windows_NT)
+	OSFLAG = $(WINDOWS)
+else
+	UNAME_S := $(shell uname -s)
+	ifeq ($(UNAME_S),Linux)
+		OSFLAG = $(LINUX)
+	endif
+	ifeq ($(UNAME_S),Darwin)
+		OSFLAG = $(OSX)
+	endif
+endif
+
 download:
 	go mod download
 
+# Note for linux users running this command, should be ran inside of a nix shell
+#  for example: nix develop -c make install
 install:
-	go get github.com/onsi/ginkgo/v2/ginkgo/generators@v2.1.2
-	go get github.com/onsi/ginkgo/v2/ginkgo/internal@v2.1.2
-	go get github.com/onsi/ginkgo/v2/ginkgo/labels@v2.1.2
-	go install github.com/onsi/ginkgo/v2/ginkgo
+ifeq ($(OSFLAG),$(WINDOWS))
+		echo "If you are running windows and know how to install what is needed, please contribute by adding it here!"
+		exit 1
+endif
+ifeq ($(OSFLAG),$(OSX))
+		brew install asdf
+		asdf plugin-add golang https://github.com/kennyp/asdf-golang.git || true
+		asdf plugin-add nodejs https://github.com/asdf-vm/asdf-nodejs.git || true
+		asdf plugin-add ginkgo https://github.com/jimmidyson/asdf-ginkgo.git || true
+		asdf install
+endif
+ifeq ($(OSFLAG),$(LINUX))
+		# pulls the ginkgo version from the .tool-versions asdf file to install so we only have the version in one place
+		go install github.com/onsi/ginkgo/v2/ginkgo@v$(shell cat ./.tool-versions | grep ginkgo | sed -En "s/ginkgo.(.*)/\1/p")
```

Comment:
You could have it install `nix` then start a shell 😄

Reply:
Ah, you could solve this on the CI by adding `ginkgo` to `shell.nix` but the version might not match: https://github.com/NixOS/nixpkgs/blob/634141959076a8ab69ca2cca0f266852256d79ee/pkgs/development/tools/ginkgo/default.nix#L5

Reply:
We want the ginkgo version to match in both places since ginkgo is good at breaking things between versions, for example currently something doesn't work right with 2.1.3 which is why we are pinned to 2.1.2. If we can keep versions in one place that would definitely be preferable.